columbo/_interaction.py

```diff
@@ -367,15 +380,31 @@ def __init__(
     def default(self) -> StaticOrDynamicValue[str]:
         return self._default
 
-    def validate(self, value: str, answers: Answers) -> Optional[str]:
+    def validate(self, value: str, answers: Answers) -> ValidationResponse:
         if self._validator is None:
             return None
         if callable(self._validator):
-            error = self._validator(value, answers)
-            if error:
-                return error
-            return None
-        raise ValueError(f"Invalid value for should_ask: {self._validator}")
+            result = self._validator(value, answers)
+            if isinstance(result, (ValidationSuccess, ValidationFailure)):
+                return result
```

Comment:
This handles the new validation responses (`ValidationFailure` and `ValidationSuccess`)